lib/cher/third_party_timeout.go

```diff
@@ -0,0 +1,21 @@
+package cher
+
+import (
+	"errors"
+)
+
+type timeoutError interface {
+	Error() string
+	Timeout() bool
+}
+
+func CoerceThirdPartyTimeout(err error) error {
+	var netErr timeoutError
+	if errors.As(err, &netErr) && netErr.Timeout() {
```

Comment:
Looks good. This timeoutError purpose is to match the net package timeout right?